### PR TITLE
Add function to return session handle from nisync

### PR DESCRIPTION
**Pull request by wchung-ni**
Signed-off-by: wchung <wai.kin.chung@ni.com>

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisync-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Add a function that return read-only session handle.

### Why should this Pull Request be merged?

Existing function that returning session object cannot fulfill NI-Digital use case.

### What testing has been done?
PR Build 
[manual pytest]
1. Added and ran a new regression test for session_handle function.
2. Ran command "poetry run ni-python-styleguide lint" and "poetry run ni-python-styleguide fix" to correct the code format as one of the requirement in CONTRIBUTING.md.
image

### CODE REVIEW

**Comment by wchung-ni** [comment]
@sjasonsmith could you review on this new add function ?

**Review by CHrislian-NI — APPROVED** [review]

**Comment by sjasonsmith** [comment]
Are you able to add a test in this same PR?

I don't think this needs very extensive testing, primarily to ensure that property remains active and doesn't go away or get renamed with future changes.

**Comment by sjasonsmith** [comment]
I don't have permissions in this repo to be able to actually review, I'm only able to comment.

File: nisync/session.py [hunk 1]
```diff
@@ -864,6 +864,10 @@ def time_references(self):
     def resource_name(self):
         return self._resource_name
 
+    @property
+    def session_handle(self):
```
Comment (sjasonsmith) [inline nisync/session.py:868]: Please add a simple regression test to ensure this doesn't change or go away in future changes.

**Review by sjasonsmith — CHANGES_REQUESTED** [review]

File: nisync/session.py [hunk 2]
```diff
@@ -864,6 +864,10 @@ def time_references(self):
...
+        return _visatype.ViSession(self._vi)
```
Comment (ni-jfitzger) [inline nisync/session.py:869]: We should just return self._vi.
Let clients worry about converting it to a ctype, if they need to.

nimi-python should automatically generate the code for conversion to ctypes in the LibararyInterpreter. For grpc, if we support it in the future, we don't pass ctypes.

File: nisync/session.py (hunk 1 above)
Comment (wchung-ni) [reply nisync/session.py:868]: Added a regression test function named as "def test___session___compare_session_handle___succeeds(lib_mock)" in test_session.py

File: nisync/session.py (hunk 2 above)
Comment (wchung-ni) [reply nisync/session.py]: I removed the parsing in the function.

**Review by tsengguanyung — APPROVED** [review]

File: tests/unit/test_session.py [hunk 3]
```diff
@@ -1,195 +1,207 @@
...
+    assert lib_mock.mock_calls == expected_calls
+
+
+def test___session___disconnect_clock_terminals___succeeds(lib_mock):
+    lib_mock.niSync_DisconnectClkTerminals.return_value = 0
+
+    with nisync.Session(RESOURCE_NAME) as session:
+        session.disconnect_clock_terminals(source=OSCILLATOR, destination=PXI_CLK10_IN)
+
+    expected_calls = [
```
Comment (sjasonsmith) [inline tests/unit/test_session.py:188]: Do you know why every line in this file shows as modified? Did the line endings change or something?

**Review by sjasonsmith — APPROVED** [review]

**Review by ni-jfitzger — APPROVED** [review]

**Review by CHrislian-NI — APPROVED** [review]

File: tests/unit/test_session.py (hunk 3 above)
Comment (wchung-ni) [reply tests/unit/test_session.py:188]: I ran command "poetry run ni-python-styleguide lint" and "poetry run ni-python-styleguide fix" to fulfill the requirement in CONTRIBUTING.md.  

The [last change](https://github.com/ni/nisync-python/pull/18/commits/693777cbdbf96b08fca9c0d1ad800a8f1f674f04) in this PR is the results of running "poetry run ni-python-styleguide fix".

**Pull request merged**